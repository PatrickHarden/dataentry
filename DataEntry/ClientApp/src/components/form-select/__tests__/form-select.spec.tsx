import React from 'react';
import { mount, shallow } from 'enzyme';

import FormSelect from '../form-select';

describe('components', () => {

  describe('form-select', () => {

    let props:any;
    let wrapper:any;

    beforeEach(() => {

        props = {
            name: "testme",
            label: "Select Me",
            defaultValue: "Dog",
            prompt: "Select somethin...",
            error: false,
            disabled: false,
            changeHandler: jest.fn(),
            options: [
                {
                    label: "Buffalo",
                    value: "Buffalo",
                    order: 4
                },
                {
                    label: "Cat",
                    value: "Cat",
                    order: 3
                },
                {
                    label: "Dog",
                    value: "Dog",
                    order: 2
                },
                {
                    label: "Parrot",
                    value: "Parrot",
                    order: 1
                },
                {
                    label: "Rat",
                    value: "Rat",
                    order: 5
                }
            ]
        }
    });

    // has proper label
    it("shows its label properly", () => {
        wrapper = mount(<FormSelect {...props}/>);
        expect(wrapper.find('h5').text()).toEqual("Select Me");
    })

    // properly sets its value
    it("sets its value correctly", () => {
        wrapper = mount(<FormSelect {...props}/>);
        expect(wrapper.find('.is-selected'));
    });

    // it has the correct prompt (always first option, when set)
    it("has a prompt", () => {
        wrapper = mount(<FormSelect {...props}/>);
        expect(wrapper.find('.Dropdown-option'))
    });

    // snapshot render
    it('renders according to snapshot', () => {
        expect(shallow(<FormSelect {...props} />)).toMatchSnapshot();
    });
  });
});