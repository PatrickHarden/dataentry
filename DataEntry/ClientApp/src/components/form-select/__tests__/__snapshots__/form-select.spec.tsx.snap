// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components form-select renders according to snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
